includes/class-smaily-for-wp-options.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * This class is used to work with the plugin's options
+ * that take user input e.g API credentials, form settings.
+ *
+ * @since      3.0.0
+ * @package    Smaily_For_WP
+ * @subpackage Smaily_For_WP/includes
+ */
+class Smaily_For_WP_Options {
+
+	/**
+	 * Smaily API credentials
+	 *
+	 * @since  3.0.0
+	 * @access private
+	 * @var    array   $api_credentials Smaily API credentials.
+	 */
+	private $api_credentials;
+
+	/**
+	 * Newsletter signup form settings.
+	 *
+	 * @since  3.0.0
+	 * @access private
+	 * @var    array   $form_options Newsletter signup form settings.
+	 */
+	private $form_options;
+
+	/**
+	 * Get API credentials.
+	 *
+	 * @since  3.0.0
+	 * @return array   $api_credentials Smaily API credentials
+	 */
+	public function get_api_credentials() {
+		if ( is_null( $this->api_credentials ) ) {
+			$this->api_credentials = $this->get_api_credentials_from_db();
+		}
+		return $this->api_credentials;
+	}
+
+	/**
+	 * Get form options.
+	 *
+	 * @since  3.0.0
+	 * @return array   $form_options Newsletter signup form settings.
+	 */
+	public function get_form_options() {
+		if ( is_null( $this->form_options ) ) {
+			$this->form_options = $this->get_form_options_from_db();
+		}
+		return $this->form_options;
+	}
+
+	/**
+	 * Get API credentials stored in database.
+	 *
+	 * @since  3.0.0
+	 * @access private
+	 * @return array   API credentials in proper format.
+	 */
+	private function get_api_credentials_from_db() {
+		$credentials = get_option( 'smailyforwp_api_option', array() );
+		return array_merge(
+			array(
+				'subdomain' => '',
+				'username'  => '',
+				'password'  => '',
+			),
+			$credentials
+		);
+	}
+
+	/**
+	 * Get form options stored in database.
+	 *
+	 * @since  3.0.0
+	 * @access private
+	 * @return array   Form options in proper format
+	 */
+	private function get_form_options_from_db() {
+		$form_options = get_option( 'smailyforwp_form_option', array() );
+		return array_merge(
+			array(
+				'form'        => '',
+				'is_advanced' => '',
+			),
+			$form_options
+		);
+	}
+
+	/**
+	 * Overwrite API credentials entry in database with provided parameter.
+	 * Disable auto-loading as API credentials are delicate.
+	 *
+	 * @since 3.0.0
+	 * @param array $api_credentials Smaily API credentials.
+	 */
+	public function update_api_credentials( $api_credentials ) {
+		// Update_option will sanitize input, before saving. We should sanitize aswell.
```

Comment:
"aswell" should be written as "as well".

Reply:
cd6cb26

Reply:
cd6cb26